### PR TITLE
Added messages to print when vane starts execution, also added progress bar for report generation

### PR DESCRIPTION
**Pull request by rewati-arista**
# Please include a summary of the changes

*pyproject.toml: added the dependency of "tqdm" used for progress bar
* report_client.py: added logic to show a progress bar when reports are being generated

Added the progress bar because as we realised with ps nrfu, sometimes reporting can take longer so this bar gives the operator a good idea of how much time to expect

* tests_tools.py, vane_cli.py, nrfu_client.py : added/refactored print messages to show while Vane is running

# Any specific logic/part of code you need extra attention on

Logic which adds progress bar for report generation currently calculates progress on the basis of dut sections written(same way pytest does), let me know how that seems

# Include the Issue number and link

#560 

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [X] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [ ] Other (please specify)

# Effort required on reviewers end

- - [ ] Easy
- - [X] Medium
- - [ ] Hard 

# How Has This Been Tested?

https://github.com/aristanetworks/vane/assets/123415500/f90d2e14-5856-43a7-8987-382eb33ecd80

# CI pipeline result

- - [X] Pass
- - [ ] Fail

### CODE REVIEW

File: vane/report_client.py [hunk 1]
```diff
@@ -819,21 +825,38 @@ def _write_detail_report(self):
 
         test_suites = self._results_datamodel["test_suites"]
 
+        # Calculating total dut sections to be written and using that as the mark for progress
```
Comment (rewati-arista) [inline vane/report_client.py:828]: github added the comparison in a complicated way, basically in the logic here, first i calculate how many dut sections be written (test suite * test cases in that test suite * duts in the test case) and later on use that number for progress calculation

**Review by kgrozis-arista — APPROVED** [review]

**Review by grybak-arista — APPROVED** [review]
This looks great!
👍🏻

**Comment by grybak-arista** [comment]
@rewati-arista This PR is ready to merge but has conflicts in the pyproject file.

**Comment by rewati-arista** [comment]
> @rewati-arista This PR is ready to merge but has conflicts in the pyproject file.

addressed that 👍

**Pull request merged**